### ISSUE TITLE
Add support for Lazy Data Provider

### ISSUE THREAD

**Issue by brunoagretti** [FlowingCode/TwinColGridAddon#130]
It'd be great if we could use lazy data providers, just like in regular grids.

**Comment by javier-godoy** [comment]
This requires a major refactoring (hence it's a good candidate for Version 3). The main issue is that the implementation assumes that the grid has a `ListDataProvider`. With a lazy data provider (in the "available" grid), the selected items would have to be filtered out, instead of removed.

**Comment by javier-godoy** [comment]
I've just realized that filtering would also provide a nice solution for the scenario discussed in the linked issue (even with ListDataProvider).

**Comment by mlopezFC** [comment]
One thing to take into account. Removing the items in the left grid might be impossible, so one idea would be to display them greyed out.

**Comment by flang** [comment]
Does it make sense to support lazy providers on selected grid? IMO only available grid should support it

**Comment by higgsja** [comment]
On the surface that makes sense. Wonder if it actually matters in any 
way, though.

a

--


Thanks!
West Point 1976 H-3
	Join us! <https://hlh-tx.com> 	*Mission:* To measurably improve 
investor performance

"We enjoy the comfort of opinion without the discomfort of thought." -- 
J.F. Kennedy

**Comment by javier-godoy** [comment]
> Does it make sense to support lazy providers on selected grid? IMO only available grid should support it.

I agree. We can start by supporting lazy data providers on the available grid (which is already a big change), and discuss lazy providers on selected grid later. I'm not sure if there are use cases where supporting so many selected items is good UX.
